tests/test_h2_client.c

```diff
@@ -1948,6 +2433,106 @@ TEST_CASE(h2_client_stream_receive_rst_stream_after_complete_response_ok) {
     return s_tester_clean_up();
 }
 
+TEST_CASE(h2_client_stream_err_input_stream_failure) {
+    ASSERT_SUCCESS(s_tester_init(allocator, ctx));
+
+    /* get connection preface and acks out of the way */
+    ASSERT_SUCCESS(h2_fake_peer_send_connection_preface_default_settings(&s_tester.peer));
+    testing_channel_drain_queued_tasks(&s_tester.testing_channel);
+    ASSERT_SUCCESS(h2_fake_peer_decode_messages_from_testing_channel(&s_tester.peer));
+
+    /* get request ready */
+    struct aws_http_message *request = aws_http_message_new_request(allocator);
+    ASSERT_NOT_NULL(request);
+
+    struct aws_http_header request_headers_src[] = {
+        DEFINE_HEADER(":method", "POST"),
+        DEFINE_HEADER(":scheme", "https"),
+        DEFINE_HEADER(":path", "/"),
+    };
+    aws_http_message_add_header_array(request, request_headers_src, AWS_ARRAY_SIZE(request_headers_src));
+
+    /* use a stalled body-stream so our test can send the response before the request is completely sent */
+    const char *body_src = "hello";
+    struct aws_byte_cursor body_cursor = aws_byte_cursor_from_c_str(body_src);
+    struct aws_input_stream *request_body = aws_input_stream_new_tester(allocator, body_cursor);
+    aws_http_message_set_body_stream(request, request_body);
+    aws_input_stream_tester_set_max_bytes_per_read(request_body, 1);
```

Comment:
[trivial] I would simplify this test and just call `aws_input_stream_tester_set_reading_broken(request_body, true /*is_broken*/);` right away, don't bother with `aws_input_stream_tester_set_max_bytes_per_read(request_body, 1)` and manually running the event-loop one tick at a time

also, the comment up there about "so our test can send the response before the request is completely sent" is copy/paste from somewhere else